src/libcore/slice.rs

```diff
@@ -897,6 +911,24 @@ impl<'a, T> Iter<'a, T> {
     ///
     /// This has the same lifetime as the original slice, and so the
     /// iterator can continue to be used while this exists.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// // First, we declare a type which implements `Iter` trait (&[usize here]):
+    /// let slice = &[1, 2, 3];
+    ///
+    /// // Then, we get the iterator:
+    /// let mut iter = slice.iter();
```

Comment:
I was wondering if I should work with blocks (like same "theme" and then empty line). But putting empty line between all lines seems good to me.